_posts/2020-11-09-pcr-primer-calculator-using-r.md

```diff
@@ -0,0 +1,295 @@
+    - if the sequence has ambiguous bases:
+      - [Yes] calculate the minimum and maximum melting temperature
+      - [No] calculate the melting temperature
+    - create variables to hold the gc-percentage and annealing primer length
+    - create a data frame using the variables we created
+4. Export the results of the calculator to a folder as a new csv file
+  - use readr to write the data frame to a csv file.
+    - save the file using the row number and primer name
+  - close the For Loop
+
+Lets take a look at the expected input file for our script.  
+![input file layout]({{ site.url }}/assets/pcr-primer-calculator-using-r-input-file.jpg)
+
+For each row with sequencing information, we want to process and output the results of PCR primer.  
+![output folder layout]({{ site.url }}/assets/pcr-primer-calculator-using-r-output-folder.jpg)
+
+Make a note about each column. Reference the column inside the code, so that we can denote each code block and what field that code relates to.  
+![output file results]({{ site.url }}/assets/pcr-primer-calculator-using-r-output-file-results.jpg)
+
+Each output file will contain the entire PCR primer sequence (leading strand, restriction site and primer sequence), a count of each nucleotide of the *annealing* portion of the primer sequence, and the melting temperature of the primer sequence. Furthermore, we need to add in a minimum and maximum melting temperature section to the output file. I will explain why we need a minimum and maximum once we begin coding the nucleotide count code.
+
+-------------------------------------------------------------------------------
+## Creating our script in R
+#### Section 1: Adding libraries to the R script  
+
+Libraries were created to make coding easier for everyone. A library inside code is synonymous to a literal library. They are full of *books*. Each book you request from the library helps you accomplish a task. Imagine yourself as a computer, your human overlord demands that you take the integral of an infinite series that is both continuous and decreasing in a positive manner and needs to know if the function converges or diverges. What are you going to do? You will go to the library and request a Calculus textbook by Ron Larson. Once you have the book, you can tell your master the area under the curve of the infinite series is finite and will converge.
+
+For our script we will be using two libraries, readr and stringr.
+```r
+# load in the packages to help simplify our code
+library("readr")
+library("stringr")
+```
+
+#### Section 2: Import the PCR primer data
+
+**Overview**  
+2. **Import the PCR primer data**  
+  - Use readr to access the primer data from a csv files  
+  - Defensive coding: check the file for primer data   
+    - If no primers are found in csv file, abort the script  
+
+The readr library handles all the necessary coding for *streaming* in the input file into our script. Now we can interacting with our primer data!
```

Comment:
This is not quite true. It reads it all in at once instead of streaming through it like command line tools (awk, sed, grep).